src/multi-selector.js

```diff
@@ -77,6 +78,15 @@ const MultiSelector = React.createClass({
 		})
 	},
 
+	shouldComponentUpdate: function(nextProps, nextState) {
+		return nextState.searchValue !== this.state.searchValue
+			|| nextProps.initialSelectedItems !== this.state.selectedItems
+			|| nextProps.items !== this.props.items
+			|| nextState.dialogDisplayed !== this.state.dialogDisplayed
+			|| nextState.activeIndex !== this.state.activeIndex
+			|| nextState.mouseIndex !== this.state.mouseIndex;
```

Comment:
I'm not sure what the default behavior is for shouldComponentUpdate when dealing with Classes (as opposed to stateless function components). What you have here is definitely better than no shouldComponentUpdate at all.